app/controllers/assessment/assessments_controller.rb

```diff
@@ -102,6 +106,77 @@ def index
     end
   end
 
+  # Export all completed assessments of the given type as a zip file of PDFs.
+  def dump_pdfs
+    assessment_type = params[:type]
+
+    # Abort if PDF export is not enabled.
+    @pdf_export = @course.pdf_export(assessment_type).display
+    if !@pdf_export
+      redirect_to access_denied_path, alert: "PDF export for #{assessment_type.pluralize} has not been enabled."
+      return
+    end
+
+    # Configure PDF generation options.
+    load_settings_for_printing
+
+    # Create temporary directory to store PDFs.
+    Dir.mktmpdir do |dir|
+
+      # For each assessment that has been completed, generate PDF and save to temporary directory. 
+      assessments = @course.assessments.send(assessment_type)
+      submissions = @course.submissions.where(std_course_id: curr_user_course.id)
+      submissions_by_assessment_id = {}
+      submissions.each do |s|
+        submissions_by_assessment_id[s.assessment_id] = s
+      end
+      assessments.each do |a|
+        @assessment = a
```

Comment:
I am sure this assessment do not need to be a instance variable as it is in a for loop.